tests/test_ensemble.py

```diff
@@ -0,0 +1,158 @@
+        self.assertAlmostEqual(ens.P, 2.0)
+        self.assertEqual(ens.V, None)
+        self.assertDictEqual(ens.mu, {'A':0.1,'B':0.2})
+        self.assertDictEqual(ens.N, {'A':None,'B':None})
+        self.assertCountEqual(ens.conjugates, ('V','N_A','N_B'))
+        self.assertAlmostEqual(ens.beta, 0.1)
+
+        #conjugates not set, V and N set
+        ens = relentless.Ensemble(types=('A','B'), T=20, V=3.0, N={'A':1,'B':2})
+        self.assertCountEqual(ens.types, ('A','B'))
+        self.assertCountEqual(ens.rdf.pairs, (('A','B'),('A','A'),('B','B')))
+        self.assertAlmostEqual(ens.kB, 1.0)
+        self.assertAlmostEqual(ens.T, 20)
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V, 3.0)
+        self.assertDictEqual(ens.mu, {'A':None,'B':None})
+        self.assertDictEqual(ens.N, {'A':1,'B':2})
+        self.assertCountEqual(ens.conjugates, ('P','mu_A','mu_B'))
+        self.assertAlmostEqual(ens.beta, 0.05)
+
+        #conjugates not set, mu and N used for one type each
+        ens = relentless.Ensemble(types=('A','B'), T=100, V=3.0, mu={'A':0.1}, N={'B':2})
+        self.assertCountEqual(ens.types, ('A','B'))
+        self.assertCountEqual(ens.rdf.pairs, (('A','B'),('A','A'),('B','B')))
+        self.assertAlmostEqual(ens.kB, 1.0)
+        self.assertAlmostEqual(ens.T, 100)
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V, 3.0)
+        self.assertDictEqual(ens.mu, {'A':0.1,'B':None})
+        self.assertDictEqual(ens.N, {'A':None,'B':2})
+        self.assertCountEqual(ens.conjugates, ('P','mu_B','N_A'))
+        self.assertAlmostEqual(ens.beta, 0.01)
+
+        #custom conjugates set, allowing both P and V
+        ens = relentless.Ensemble(types=('A','B'), T=100, P=1.5, V=3.0, mu={'A':0.1}, N={'B':2}, conjugates=('mu_B','N_A'))
+        self.assertAlmostEqual(ens.P, 1.5)
+        self.assertAlmostEqual(ens.V, 3.0)
+        self.assertDictEqual(ens.mu, {'A':0.1,'B':None})
+        self.assertDictEqual(ens.N, {'A':None,'B':2})
+        self.assertCountEqual(ens.conjugates, ('mu_B','N_A'))
+
+        #custom conjugates set, allowing both mu and N
+        ens = relentless.Ensemble(types=('A','B'), T=100, V=3.0, mu={'A':0.1,'B':0.2}, N={'A':1,'B':2}, conjugates=('P',))
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V, 3.0)
+        self.assertDictEqual(ens.mu, {'A':0.1,'B':0.2})
+        self.assertDictEqual(ens.N, {'A':1,'B':2})
+        self.assertCountEqual(ens.conjugates, ('P',))
+
+        #custom conjugates set, allowing P, V, mu, and N
+        ens = relentless.Ensemble(types=('A','B'), T=100, P=1.5, V=3.0, mu={'A':0.1,'B':0.2}, N={'A':1,'B':2}, conjugates=('q',))
+        self.assertEqual(ens.P, 1.5)
+        self.assertAlmostEqual(ens.V, 3.0)
+        self.assertDictEqual(ens.mu, {'A':0.1,'B':0.2})
+        self.assertDictEqual(ens.N, {'A':1,'B':2})
+        self.assertCountEqual(ens.conjugates, ('q',))
```

Comment:
So here are all examples where `conjugates` isn't "correct", in that it doesn't specify all the right conjugate variables.  Might need to change behavior depending on what we decided for my earlier comment on this issue.